app/services/messages/mention_service.rb

```diff
@@ -19,14 +19,35 @@ def valid_mention_message?(message)
   end
 
   def mentioned_ids
-    @mentioned_ids ||= message.content.scan(%r{\(mention://(user|team)/(\d+)/(.+?)\)}).map(&:second).uniq
+    user_mentions = message.content.scan(%r{\(mention://user/(\d+)/(.+?)\)}).map(&:first)
+    team_mentions = message.content.scan(%r{\(mention://team/(\d+)/(.+?)\)}).map(&:first)
+
+    expanded_user_ids = expand_team_mentions_to_users(team_mentions)
+
+    (user_mentions + expanded_user_ids).uniq
+  end
+
+  def expand_team_mentions_to_users(team_ids)
+    return [] if team_ids.blank?
+
+    valid_teams = message.inbox.account.teams
```

Comment:
we don't need to do the join with with inbox here. since we are doing the filtering later in `filter_mentioned_ids_by_inbox`